pkgs/by-name/as/ascii-draw/package.nix

```
@@ -0,0 +1,66 @@
+{ lib
+, python3Packages
+, fetchFromGitHub
+, meson
+, ninja
+, pkg-config
+, gobject-introspection
+, wrapGAppsHook4
+, desktop-file-utils
+, libadwaita
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "ascii-draw";
+  version = "0.3.0";
+  pyproject = false;
+
+  src = fetchFromGitHub {
+    owner = "Nokse22";
+    repo = "ascii-draw";
+    rev = "v${version}";
+    hash = "sha256-vI+j8OuQ3b6La0+7wWeoUtBal24dazlN/T0Bng5TgMo=";
+  };
+
+  # Silently fail autosaving to flatpak directory
+  # Autosaving won't work, but program runs normally otherwise
+  # https://github.com/Nokse22/ascii-draw/issues/31
+  postPatch = ''
+    substituteInPlace src/window.py \
+      --replace-fail "/var/data" ""
+  '';
+
+  nativeBuildInputs = [
+    meson
+    ninja
+    pkg-config
+    gobject-introspection
+    wrapGAppsHook4
+    desktop-file-utils
+  ];
+
+  buildInputs = [
+    libadwaita
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
+    pygobject3
+    pyfiglet
+    emoji
+  ];
+
+  dontWrapGApps = true;
+
+  preFixup = ''
+    makeWrapperArgs+=("''${gappsWrapperArgs[@]}")
+  '';
+
+  meta = {
+    description = "An app to draw diagrams or anything using only ASCII";
+    homepage = "https://github.com/Nokse22/ascii-draw";
+    license = lib.licenses.gpl3Plus;
+    mainProgram = "ascii-draw";
+    maintainers = with lib.maintainers; [ aleksana ];
+    platforms = lib.platforms.unix;
```

```suggestion
    platforms = lib.platforms.linux;
```

https://www.gnu.org/software/gnulib/manual/html_node/bindtextdomain.html

---

oops